articles/quickstart/native/ionic-angular/01-login.md

```diff
@@ -0,0 +1,245 @@
+* Specify the configuration for the Auth0 Angular SDK
+
+```javascript
+// Import the types from the SDK
+import { AuthModule } from '@auth0/auth0-angular';
+import config from '../../capacitor.config';
+
+// ..
+
+// Build the URL that Auth0 should redirect back to
+const redirectUri = `<%= "${config.appId}" %>://${account.namespace}/capacitor/<%= "${config.appId}" %>/callback`;
+
+// Register AuthModule with your AppModule
+@NgModule({
+  declarations: [AppComponent],
+  entryComponents: [],
+  imports: [
+    BrowserModule,
+    IonicModule.forRoot(),
+    AppRoutingModule,
+    AuthModule.forRoot({
+      domain: "${account.namespace}",
+      clientId: "${account.clientId}",
+      redirectUri,
+      cacheLocation: 'localstorage',
+      useRefreshTokens: true,
+    }),
+  ],
+  providers: [{ provide: RouteReuseStrategy, useClass: IonicRouteStrategy }],
+  bootstrap: [AppComponent],
+})
+```
+
+The `AuthModule.forRoot` function takes the following configuration:
+
+- `domain`: The "domain" value present under the "Settings" of the application you created in your Auth0 dashboard, or your custom domain if using Auth0's [Custom Domains feature](http://localhost:3000/docs/custom-domains)
+- `clientId`: The "client ID" value present under the "Settings" of the application you created in your Auth0 dashboard
+- `redirectUri`: The URL to where you'd like to redirect your users after they authenticate with Auth0.
+- `useRefreshTokens`: The Angular SDK will use refresh tokens over relying on third-party cookies, which can be blocked in many modern browsers. Please read [Refresh Token Rotation](https://auth0.com/docs/tokens/refresh-tokens/refresh-token-rotation) for more information on using refresh tokens with public clients.
+- `cacheLocation`: The location at which to store tokens. We use `localstorage` here so that tokens are persisted across app refreshes.
+
+<%= include('../_includes/ionic/_note_storage') %>
+
+:::panel Checkpoint
+Now that you have configured your app with the Auth0 Angular SDK, run your application to verify that the SDK is initializing without error and that your application runs as it did before.
+:::
+
+## Add Login to Your Application
+
+<%= include('../_includes/ionic/_add_login_intro') %>
+
+Add a new `LoginButton` component to your application with the following code:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+import { Browser } from '@capacitor/browser';
+import { tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-login-button',
+  template: `<ion-button (click)="login()">Login</ion-button>`,
+})
+export class LoginButtonComponent {
+  constructor(public auth: AuthService) {}
+
+login() {
+    this.auth
+      .buildAuthorizeUrl()
+      .pipe(tap((url) => Browser.open({ url, windowName: '_self' })))
+      .subscribe();
+  }
```

Comment:
What do you think about waiting for the promise? `Browser.open()` returns a promise, however, every code a user ever puts after the `tap` will be executed before the promise is resolved, while people might expect differently.

This can be solved using 

```
login() {
    this.auth
      .buildAuthorizeUrl()
      .pipe(mergeMap((url) => Browser.open({ url, windowName: '_self' })))
      .subscribe();
  }
```